### ISSUE TITLE
Second Screen WG Feedback: synchronous APIs that require user input are problematic

### ISSUE THREAD

**Issue by travisleithead** [w3c/html#163]
This is feedback provided from the W3C Second Screen WG to the @w3c/webplatform team generally. I'm filing it as an issue on HTML as several of the impacted APIs are defined in the HTML spec, but this is a general issue for consideration.

Paraphrased:

> We [the Second Screen WG] identified a somewhat interesting issue that is manifest when you start launching browsing contexts on screens where there is either no facility for processing user input, or no user nearby (i.e., a second screen connection to a "smart" projector). In these cases, some parts of the web platform are not workable when no user is available. `window.alert()` is one such example--a blocking API that requires a user to dismiss the UI. Another example is the `<input type=file>' when its`click()` API is called. (We don't expect this problem is getting worse, since newer APIs that interact with the user are generally designed to be asynchronous and won't block the platform when the user doesn't respond or isn't present.) We felt this was an opportunity for collaboration between the Web Platform WG and the SSWG.

See also:
- TAG feedback summary: https://github.com/w3ctag/spec-reviews/issues/61#issuecomment-152439245
- Related Issue on Presentation API: https://github.com/w3c/presentation-api/issues/99

While the SSWG is not blocked on this issue, it would be nice [generally] to describe some conditional behavior for how these APIs should behave in environments where the user isn't present.

**Comment by siusin** [comment]
We're closing this issue on the W3C HTML specification because the W3C and WHATWG are now [working together on HTML](https://www.w3.org/blog/2019/05/w3c-and-whatwg-to-work-together-to-advance-the-open-web-platform/), and all issues are being discussed on the [WHATWG repository](https://github.com/whatwg/html).

If you filed this issue and you still think it is relevant, please open a [new issue](https://github.com/whatwg/html/issues) on the WHATWG repository and reference this issue (if there is useful information here). Before you open a new issue, please check for existing issues on the WHATWG repository to avoid duplication.

If you have questions about this, please open an issue on the [W3C HTML WG repository](https://github.com/w3c/htmlwg/issues/new) or send an email to public-html@w3.org.
